core/src/health.rs

```diff
@@ -0,0 +1,83 @@
+//! Module implementing shared basic health reporting.
+
+use crate::http_server::Handler;
+use anyhow::Result;
+use rouille::{Request, Response};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+/// Trait for asyncronously notifying health information.
+#[cfg_attr(test, mockall::automock)]
+pub trait HealthReporting: Send + Sync {
+    /// Notify that the service is ready.
+    fn notify_ready(&self);
+}
+
+/// Implementation sharing health information over an HTTP endpoint.
+#[derive(Debug, Default)]
+pub struct HttpHealthEndpoint {
+    ready: AtomicBool,
+}
+
+impl HttpHealthEndpoint {
+    /// Creates a new HTTP health enpoint.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Returns true if the service is ready, false otherwise.
+    fn is_ready(&self) -> bool {
+        self.ready.load(Ordering::Acquire)
```

Comment:
So fancy, why not just `SeqCst`?

Reply:
Acquire, Release is more efficient and sufficient in this case even though for this operation the speed difference doesn't matter.

Reply:
So, from the docs:
>
> all previous writes become visible to all threads that perform an `Acquire` (or stronger) load of this value.

Reply:
No strong feelings. I felt that using the "right" flags here requires some thought and care which isn't really necessary given the marginal benefit. But if you prefer to be "optimal", that's fine.

Reply:
Switched it to `SeqCst` for lower developer overhead.